DataDog/chaos-controller
# Make optional fields nullable

### What does this PR do?

It marks optional fields having a special type (basically `object` and `array` types) as nullable. It also removes the `kubebuilder:validation:Optional` as it is not required when specifying the `omitempty` JSON annotation.

### Motivation

If not set as nullable, some fields can't be not specified even if marked as optional. Here's an example error when not specifying the `hosts` field for the `networkLatency` disruption:

```
2020-03-22T01:39:32.652Z        ERROR   controller-runtime.controller   Reconciler error        {"controller": "disruption", "request": "default/disruption-sample", "error": "Disruption.chaos.datadoghq.com \"disruption-sample\" is invalid: spec.networkLatency.hosts: Invalid value: \"null\": spec.networkLatency.hosts in body must be of type array: \"null\""}
```

### Testing Guidelines

* install the new CRD with `make install`
* apply the disruption sample without specifying the `hosts` field in `networkLatency`

It should not error and keep the field as `nil`.

### Additional Notes

The PR also fixes a small typo in a log call.
